Http/Yclient.php

```diff
@@ -115,16 +116,24 @@ private function doRequest(
             $logData[] = 'response = ' . $responseContent;
             $this->yotpoApiLogger->info($logMessage, $logData);
         } catch (GuzzleException $exception) {
-            /** @var Response $response */
-            $response = $this->responseFactory->create([
-                'status' => $exception->getCode(),
-                'reason' => $exception->getMessage()
-            ]);
             $exceptionData = [];
             $exceptionMessage = 'API Error';
             $exceptionData[] = 'API URL = ' . $baseUrl . $uriEndpoint;
             $exceptionData[] = $options;
-            $exceptionData[] = 'response code = ' . $exception->getCode();
+            try {
+                $response = $this->responseFactory->create([
+                    'status' => $exception->getCode(),
+                    'reason' => $exception->getMessage()
+                ]);
+                $exceptionCode = $exception->getCode();
+            } catch (\InvalidArgumentException $exception) {
```

Comment:
Why do we need this extra try catch clause? this catch clause will never be reached.

Reply:
https://github.com/YotpoLtd/magento2-module-core/pull/177#discussion_r834104296

Reply:
If what Manaf suggesting is a real use case, then still this try catch is incorrect, it should be like this:

```suggestion
        try { 
        -- tried code here --
        } catch (GuzzleException $exception) {
            /** @var Response $response */
            $response = $this->responseFactory->create([
                'status' => $exception->getCode(),
                'reason' => $exception->getMessage()
            ]);
            $exceptionData = [];
            $exceptionData[] = 'API URL = ' . $baseUrl . $uriEndpoint;
            $exceptionData[] = $options;
            $exceptionData[] = 'response code = ' . $exception->getCode();
            $response = $this->responseFactory->create([
                    'status' => $exception->getCode(),
                    'reason' => $exception->getMessage()
            $exceptionCode = $exception->getCode();

            $exceptionData[] = 'response = ' . $exception->getMessage();
            $exceptionMessage = 'API Error';
            $this->yotpoApiLogger->info($exceptionMessage, $exceptionData);
            } catch (\InvalidArgumentException $exception) {
             /** @var Response $response */
            $response = $this->responseFactory->create([
                'status' => $exception->getCode(),
                'reason' => $exception->getMessage()
            ]);
            $exceptionData = [];
            $exceptionMessage = 'API Error';
            $exceptionData[] = 'API URL = ' . $baseUrl . $uriEndpoint;
            $exceptionData[] = $options;
            $exceptionData[] = 'response code = ' . $exception->getCode();
            $response = $this->responseFactory->create([
                    'status' => $exception->getCode(),
                    'reason' => $exception->getMessage()
            $exceptionCode = $exception->getCode();

            $exceptionData[] = 'response = ' . $exception->getMessage();
            $this->yotpoApiLogger->info($exceptionMessage, $exceptionData);
            }
```

Reply:
Following our offline discussion, here's the source code - https://github.com/guzzle/psr7/blob/master/src/Response.php